### ISSUE TITLE
librdkafka can't be built on Solaris 10

### ISSUE THREAD

**Issue by guogang-li** [edenhill/librdkafka#732]
Description
===========
rdgz.c: In function 'rd_gz_decompress':
rdgz.c:53: error: 'gz_header' undeclared (first use in this function)
rdgz.c:53: error: (Each undeclared identifier is reported only once
rdgz.c:53: error: for each function it appears in.)
rdgz.c:53: error: expected ';' before 'hdr'
rdgz.c:65: warning: implicit declaration of function 'inflateGetHeader'
rdgz.c:65: error: 'hdr' undeclared (first use in this function)
make[1]: *** [../mklove/Makefile.base:77: rdgz.o] Error 1

zlib.h does live in /usr/include, and gz_header does exist:

typedef struct gz_header_s {
    int     text;       /* true if compressed data believed to be text */
    uLong   time;       /* modification time */
    int     xflags;     /* extra flags (not used when writing a gzip file) */
    int     os;         /* operating system */
    Bytef   *extra;     /* pointer to extra field or Z_NULL if none */
    uInt    extra_len;  /* extra field length (valid if extra != Z_NULL) */
    uInt    extra_max;  /* space at extra (only when reading header) */
    Bytef   *name;      /* pointer to zero-terminated file name or Z_NULL */
    uInt    name_max;   /* space at name (only when reading header) */
    Bytef   *comment;   /* pointer to zero-terminated comment or Z_NULL */
    uInt    comm_max;   /* space at comment (only when reading header) */
    int     hcrc;       /* true if there was or will be a header crc */
    int     done;       /* true when done reading gzip header (not used
                           when writing a gzip file) */
} gz_header;

How to reproduce
================
SunOS 5.10 sun4v sparc SUNW,SPARC-Enterprise-T5220
GNU/make/make-4.2/make
GNU/gcc/gcc-4.4.5/bin/gcc
GNU/gcc/gcc-4.4.5/bin/g++

Checklist
=========
Please provide the following information:

 - [ ] librdkafka version (release number or git tag):
 - [ ] Apache Kafka version: latest from https://github.com/edenhill/librdkafka
 - [ ] librdkafka client configuration:
 - [ ] Operating system: SunOS 5.10 
 - [ ] Using the legacy Consumer
 - [ ] Using the high-level KafkaConsumer
 - [ ] Provide logs (with `debug=..` as necessary) from librdkafka
 - [ ] Provide broker log excerpts
 - [ ] Critical issue

**Comment by edenhill** [comment]
Might be a missing  include or define, I don't have access to a Solaris box so I can't do this myself.

**Comment by edenhill** [comment]
Can you figure out what is needed to compile and file a PR?

**Comment by guogang-li** [comment]
Thank you so much for the follow up!

If I understand it correctly, the section of code that won’t compile on Solaris is related to compress and decompress, I guess we can comment it out since we may not directly us this feature.

Another issue we have is the shell script, configure, does not take the native cc and CC compilers. We tried two different ways:

configure --prefix /etsi/work/gli/kafka/librdkafka-master \
--exec-prefix=/etsi/work/gli/kafka/librdkafka-master/local \
--includedir=/usr/include \
--cc=/opt/SUNWspro/bin/cc \
--cxx=/opt/SUNWspro/bin/CC \
--mbits=32 \
--enable-static \
--disable-ssl \
--disable-sasl

Or
export CC=/opt/SUNWspro/bin/cc
export CXX=/opt/SUNWspro/bin/CC

configure --prefix /etsi/work/gli/kafka/librdkafka-master \
--exec-prefix=/etsi/work/gli/kafka/librdkafka-master/local \
--includedir=/usr/include \
--mbits=32 \
--enable-static \
--disable-ssl \
--disable-sasl

In either case, configure takes gcc and g++:

checking for OS or distribution... ok (SunOS)
checking for C compiler from CC env... failed
checking for gcc (by command)... ok
checking for C++ compiler from CXX env... failed
checking for C++ compiler (g++)... ok

Thanks again!

**Comment by edenhill** [comment]
Regarding CC/CXX, it does check your env setting but it fails for some reason, can you paste your entire config.log?

**Comment by guogang-li** [comment]
# configure exec: ./configure --prefix /etsi/work/gli/kafka/librdkafka-master --exec-prefix=/etsi/work/gli/kafka/librdkafka-master/local --includedir=/usr/include --mbits=32 --enable-static --disable-ssl --disable-sasl
# On Thu Aug  4 08:51:53 CDT 2016
DBG: ccenv: WITH_CC: /opt/SUNWspro/bin/cc --version failed: cc: Warning: option -- passed to ld
usage: cc [ options] files.  Use 'cc -flags' for details
DBG: Check ccenv (WITH_CC, action cont (originally cont)) failed: command '/opt/SUNWspro/bin/cc --version' failed:
cc: Warning: option -- passed to ld
usage: cc [ options] files.  Use 'cc -flags' for details
DBG: cxxenv: WITH_CXX: /opt/SUNWspro/bin/CC --version failed: CC: Warning: Option --version passed to ld, if ld is invoked, ignored otherwise
ld: Software Generation Utilities - Solaris Link Editors: 5.10-1.1520
…

Solaris compilers show versions in a different way:

$> /opt/SUNWspro/bin/cc -V
cc: Sun C 5.13 SunOS_sparc 2014/10/20

$> /opt/SUNWspro/bin/CC -V
CC: Sun C++ 5.13 SunOS_sparc 2014/10/20

**Comment by edenhill** [comment]
You need to build librdkafka with the GNU (gcc) or LLVM (clang) toolchains, the Sun compiler is not supported.

**Comment by guogang-li** [comment]
Thanks for your reply. I really appreciate your help!
